### PR TITLE
Merge ign-gazebo6 ➡️  main (part 3/3)

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️  Forward port

Port `ign-gazebo6 ` ➡️  `main`

Branch comparison: https://github.com/gazebosim/gz-sim/compare/main...ign-gazebo6

* Builds on top of https://github.com/gazebosim/gz-sim/pull/1631
* Part of https://github.com/gazebo-tooling/release-tools/issues/578

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by mjcarroll** [comment]
Have you verified that the demo worlds as part of the quick start GUI are functional on Garden?  I seem to recall trying one or two and they still used Scene3d.

**Comment by chapulina** [comment]
> Have you verified that the demo worlds as part of the quick start GUI are functional on Garden? I seem to recall trying one or two and they still used Scene3d.

Yeah I just checked this. I'm going to open a separate PR loading `MinimalScene` + others when `GzScene3d` is requested.

Btw, I'm updating this PR. Will ping reviewers when it's open for review again.

**Comment by chapulina** [comment]
> Have you verified that the demo worlds as part of the quick start GUI are functional on Garden? I seem to recall trying one or two and they still used Scene3d.

* See https://github.com/gazebosim/gz-sim/pull/1605

File: src/gui/Gui_TEST.cc [hunk 1]
```diff
@@ -167,3 +170,103 @@ TEST_F(GuiTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(PathManager))
...
+  bool worldsCalled{false};
+  std::function<bool(msgs::StringMsg_V &)> worldsCb =
+      [&worldsCalled](msgs::StringMsg_V &_res)
+      {
+        _res.add_data("world_name");
+        worldsCalled = true;
+        return true;
+      };
+  node.Advertise("/gazebo/worlds", worldsCb);
+  igndbg << "Worlds advertised" << std::endl;
```
Comment (jennuine) [inline src/gui/Gui_TEST.cc:192]:
```suggestion
  gzdbg << "Worlds advertised" << std::endl;
```

File: src/gui/Gui_TEST.cc [hunk 2]
```diff
@@ -167,3 +170,103 @@ TEST_F(GuiTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(PathManager))
...
+  std::function<void(const msgs::StringMsg &)> topicCb =
+      [&startingWorldCalled](const auto &_msg)
+      {
+        EXPECT_EQ("banana", _msg.data());
+        startingWorldCalled = true;
+      };
+
+  std::string topic{"/gazebo/starting_world"};
+  node.Subscribe(topic, topicCb);
+  igndbg << "Subscribed to [" << topic << "]" << std::endl;
```
Comment (jennuine) [inline src/gui/Gui_TEST.cc:205]:
```suggestion
  gzdbg << "Subscribed to [" << topic << "]" << std::endl;
```

File: src/gui/Gui_TEST.cc [hunk 3]
```diff
@@ -167,3 +170,103 @@ TEST_F(GuiTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(PathManager))
...
+  std::string configFilePath{"/tmp/quick_start_test.config"};
+  std::ofstream configFile(configFilePath);
+  configFile << "<window><dialog_on_exit>false</dialog_on_exit></window>" <<
+      "<plugin filename='Publisher' name='Publisher'/>";
+  configFile.close();
+
+  // Thread to check and close quick start dialog
+  std::thread checkingThread([&]()
+  {
+    igndbg << "Started checking thread" << std::endl;
```
Comment (jennuine) [inline src/gui/Gui_TEST.cc:218]:
```suggestion
    gzdbg << "Started checking thread" << std::endl;
```

File: src/gui/Gui_TEST.cc [hunk 4]
```diff
@@ -167,3 +170,103 @@ TEST_F(GuiTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(PathManager))
...
+        (nullptr == gui::App( ) ||
+         gui::App()->allWindows().empty() ||
+         !gui::App()->allWindows()[0]->isVisible())
+        && sleep < 30; ++sleep)
+    {
+      std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    }
+    ASSERT_NE(nullptr, gui::App());
+    ASSERT_EQ(1, gui::App()->allWindows().count());
+    igndbg << "Found app" << std::endl;
```
Comment (jennuine) [inline src/gui/Gui_TEST.cc:229]:
```suggestion
    gzdbg << "Found app" << std::endl;
```

File: src/gui/Gui_TEST.cc [hunk 5]
```diff
@@ -167,3 +170,103 @@ TEST_F(GuiTest, GZ_UTILS_TEST_ENABLED_ONLY_ON_LINUX(PathManager))
...
+  });
+
+  auto app = createGui(gg_argc, gg_argv,
+      configFilePath.c_str() /* _guiConfig */,
+      nullptr /* _defaultGuiConfig */,
+      true /* _loadPluginsFromSdf */,
+      nullptr /* _sdfFile */,
+      true /* _waitGui */);
+  EXPECT_NE(nullptr, app);
+  igndbg << "GUI created" << std::endl;
```
Comment (jennuine) [inline src/gui/Gui_TEST.cc:264]:
```suggestion
  gzdbg << "GUI created" << std::endl;
```

**Review by jennuine — COMMENTED** [review]
Found a couple more `ign`s

**Comment by chapulina** [comment]
> Found a couple more igns

Thanks, @jennuine ! Updated in https://github.com/gazebosim/gz-sim/compare/eef117a6bffac0e3e66e95dad74a6be3bdb73561..9265793caafadc6f92e79342defba11ad6a98a28

**Pull request merged**